components/eam/src/chemistry/modal_aero/modal_aero_amicphys.F90

```diff
@@ -3584,334 +3576,608 @@ end subroutine mam_soaexch_1subarea
-      do iaer = 1, naer
-         tmpd = qaer_cur(iaer,mfrm)*xferfrac_vol
-         qaer_cur(iaer,mfrm) = qaer_cur(iaer,mfrm) - tmpd
-         qaer_cur(iaer,mtoo) = qaer_cur(iaer,mtoo) + tmpd
-      end do ! iaer
-
-      if ( iscldy_subarea ) then
-      tmpd = qnumcw_cur(mfrm)*xferfrac_num
-      qnumcw_cur(mfrm) = qnumcw_cur(mfrm) - tmpd
-      qnumcw_cur(mtoo) = qnumcw_cur(mtoo) + tmpd
-      do iaer = 1, naer
-         tmpd = qaercw_cur(iaer,mfrm)*xferfrac_vol
-         qaercw_cur(iaer,mfrm) = qaercw_cur(iaer,mfrm) - tmpd
-         qaercw_cur(iaer,mtoo) = qaercw_cur(iaer,mtoo) + tmpd
-      end do ! iaer
-      end if ! ( iscldy_subarea ) then
+           call compute_size_factor (src_mode, sz_factor) !size factor for "src mode"
+           call compute_size_factor (dest_mode,   sz_factor) !size factor for "dest mode"
 
+           !---------------------------------------------------------------------------------------------------------
+           ! We compute few factors below for the "src_mode", which will be used for inter-mode particle transfer
+           !---------------------------------------------------------------------------------------------------------
 
-#if ( defined( CAMBOX_NEVER_ACTIVATE_THIS ) )
-!   diagnostic output start ----------------------------------------
-                if (ldiag1 > 0) then
-                if ((i == icol_diag) .and. (mod(k-1,5) == 0)) then
-                  if (lstooa .gt. 0) then
-                    write(lund,'(a,i4,2(2x,a),1p,10e14.6)') 'RENAME qdels', iq,   &
-                        cnst_name(lsfrma+loffset), cnst_name(lstooa+loffset),   &
-                        deltat*dqdt(i,k,lsfrma), deltat*(dqdt(i,k,lsfrma) - xfertend),   &
-                        deltat*dqdt(i,k,lstooa), deltat*(dqdt(i,k,lstooa) + xfertend)
-                  else
-                    write(lund,'(a,i4,2(2x,a),1p,10e14.6)') 'RENAME qdels', iq,   &
-                        cnst_name(lsfrma+loffset), cnst_name(lstooa+loffset),   &
-                        deltat*dqdt(i,k,lsfrma), deltat*(dqdt(i,k,lsfrma) - xfertend)
-                  end if
-                end if
-                end if
-!   diagnostic output end   ------------------------------------------
-#endif
+           fmode_dist_tail_fac(src_mode) = sqrt_half/alnsg_aer(src_mode) !factor for computing distribution tails of the  "src mode"
 
+           !compute volume to number high and low limits with relaxation coefficients (watch out for repeated calculations)
+           v2n_lo_rlx(src_mode) = compute_vol_to_num_ratio(src_mode, dgnumlo_aer) * frelax
+           v2n_hi_rlx(src_mode) = compute_vol_to_num_ratio(src_mode, dgnumhi_aer) / frelax
 
-      end do mainloop1_ipair
+           !A factor for computing diameter at the tails of the distribution
+           ln_diameter_tail_fac(src_mode) = 3.0 * (alnsg_aer(src_mode)**2)
 
+           !Cut-off (based on geometric mean) for making decision to do inter-mode transfers
```

Comment:
Is there a simple explanation for how this cutoff is determined? Not a big deal, but if it's easy to explain, this would be a good place to do so.